src/pages/QuestionListsPage/QuestionLists.tsx

```diff
@@ -1,180 +1,163 @@
+import CreateQuestionListModal from '../../domain/QuestionLists/CreateQuestionListModal';
+import MoreIconContent from '../../domain/QuestionLists/MoreIconContent';
+import styles from '../../domain/QuestionLists/QuestionLists.module.scss';
+import QuestionListsHeader from '../../domain/QuestionLists/QuestionListsHeader';
+import { useLocalStorage } from '../../hooks';
+import ScalableBody from '../../layout/scalableBody/ScalableBody';
 import {
   Client,
   CreateQuestionListRequest,
   InterviewQuestionModel,
   QuestionListModel,
 } from '../../services/Client';
-import styles from './QuestionLists.module.scss';
-
-const { TextArea } = Input;
+import { filterLists } from '../../services/filterService';
+import {
+  colorByCategory,
+  getDistinctCategories,
+} from '../../services/tagCategoryColorService';
 
 export default function QuestionLists(): JSX.Element {
-  const client = useMemo(() => new Client(), []);
+  const client = new Client();
+  const navigate = useNavigate();
   const [isPopupVisible, setPopupVisible] = useState(false);
-  const [titleInput, setTitleInput] = useState('');
-  const [descriptionInput, setDescriptionInput] = useState('');
+  const [searchText, setSearchText] = useState<string>('');
   const [lists, setLists] = useState<QuestionListModel[] | null>();
-  const colors: string[] = useMemo(
-    () => ['red', 'blue', 'green', 'orange', 'purple', 'yellow', 'black'],
+  const [recentListIds, setRecentListIds] = useLocalStorage<number[]>(
+    'recentListIds',
     []
   );
-  const colorIndex = useRef(0);
-  const hashmap = useMemo(() => new Map<string, string>(), []);
-
-  const nextColor = useCallback((): string => {
-    const color = colors[colorIndex.current];
-    colorIndex.current = (colorIndex.current + 1) % colors.length;
-    return color;
-  }, [colors]);
-
-  const colorByCategory = useCallback(
-    (category: string): string => {
-      if (!hashmap.has(category)) {
-        hashmap.set(category, nextColor());
-      }
-      return hashmap.get(category) ?? 'black';
-    },
-    [hashmap, nextColor]
-  );
-
-  function getDistinctCategories(
-    questions: InterviewQuestionModel[]
-  ): string[] {
-    const set = new Set<string>();
-    questions?.forEach((question) => {
-      if (question?.category) {
-        set.add(question.category);
-      }
-    });
-    return Array.from(set);
-  }
 
   useEffect(() => {
-    async function loadLists(): Promise<void> {
-      const data = await client.questionLists(undefined, undefined, undefined);
-
-      setLists(data);
+    client.questionLists(undefined, undefined, undefined).then((value) => {
+      setLists(value);
+    });
+  }, []);
+
+  const addListIdToRecentlyUsed = (listId: number): void => {
+    let currentIds = recentListIds;
+    if (currentIds.includes(listId)) {
+      currentIds = currentIds.filter((id) => id !== listId);
+    } else if (currentIds.length >= 5) {
```

Comment:
I would replace the `5` into constant. I am thinking why is here 5 and not 4 or 6 for example :)